### PR TITLE
PLAT-1538: REQUEST: Control the padding on the modal slots

### PR DESCRIPTION
**Pull request by chavlji**
[JIRA ticket](https://reciprocitylabs.atlassian.net/browse/PLAT-1538)

### CODE REVIEW

**Comment by homer0** [comment]
I understand where are you trying to go, but I don't think this is the best solution. If the implementation can't control the padding, the default should be _none_: based on the use cases we have right now, zencomply will use the `paddingless-` slots, but if they use them, they don't get the footer button nor the close icon.

Another thing that could do is to create the attributes `header-padding`, `content-padding` and `footer-padding` and forward them to the `zen-space` components that wrap each slot. If I'm not wrong, the only requirement right now is with the header, so we'll kind of cover our default use case and the customization needed by the implementation.

We can talk about this tomorrow :D.

Btw, a few small extra SUPER-NON-BLOCKING details:

- The icon for the modal should be inside a button or have the `role="button"`.
- Use `feat` not `feature` for the commits.
- The scope for the commits, like on `feature: zen-space - size xl to 24 px` should be wrapped in parentheses: `feat(zen-space): size xl to 24 px`.
- You are mixing `test` and `tests` on the commits.

**Comment by chavlji** [comment]
You're right. Not getting default close or ok buttons is an issue. Setting default padding to none here isn't ok I guess, because we have exact specs from designers. All components should look as per designer's specs ootb.

I like the approach with exposing extra padding props for each part! It also follows the same philosophy as all our other components (they'll support similar `padding` prop). I'll add those props to modal now.

As for commit messages, noted! Didn't know the thing of how the scope should be defined. Thanks! I'll check some conventional commits manual now.

**Review by mihaelpalfi — COMMENTED** [review]
Looks nice but code is a little complicated with all this slots. Why not just wrap header, footer, content in zen-space and add props so that we can set the paddings. Default none.

And also remembered that we also have to change the displaying of the modal like we did for notifications. Right? If yes and there is no ticket then we should create it.

**Comment by chavlji** [comment]
I agree it's too complicated. I'll change it as you suggest (expose padding of zen-space as props).

@mihaelpalfi what do you mean with "have to change the displaying of the modal like we did for notifications"? Imo the way modal currently works is ok?

**Review by mihaelpalfi — COMMENTED** [review]
Good work. Found this issue but probably not related to this issue. 
image
Padding to large

File: src/stories/changelog.stories.mdx [hunk 1]
```diff
@@ -7,6 +7,10 @@
 #### Breaking changes
 
 1. Prop `padding` value `xl` was changed from `2rem` to `1.5rem`. New prop value `2xl` was added that equals `2rem`. So to keep the same padding, replace `padding="xl"` with `padding="2xl"`
+1. `zen-modal`: For consistency reasons, the following slots were renamed:
+
+- `footer` to `paddingless-footer`
+- `buttons` to `footer`
```
Comment (homer0) [inline src/stories/changelog.stories.mdx:13]: Remember to update this.

(Btw, why did you use two `1.` :P? I've been checking docs and all the lists only use `1.` for items xD)

File: src/components/zen-modal/zen-modal.scss [hunk 2]
```diff
@@ -48,29 +46,22 @@ $show-time: 0.3s;
...
     opacity: 0;
     transform: translate(0, -30px);
   }
 }
 
 .header {
-  display: flex;
-  align-items: center;
-  padding: 1rem 0 1rem $padding-x;
+  border-bottom: 1px solid $color-gray-200;
```
Comment (homer0) [inline src/components/zen-modal/zen-modal.scss:57]: Not sure in which commit this was added, but it should have something like this:

```js
BREAKING CHANGE: The modal header now has a border bottom
```

... or similar; that's a fix from our point of view, but a breaking change for the implementation.

**Review by homer0 — APPROVED** [review]
🤘!

**Review by mihaelpalfi — APPROVED** [review]
Great animations Jan. Tested and setting the paddings works. One small NIT : 
The close button gets pushed down on long title names.

image

File: src/stories/changelog.stories.mdx (hunk 1 above)
Comment (chavlji) [reply src/stories/changelog.stories.mdx:13]: md takes care of this and converts it to correct numbers. No need to keep tracks of that manually:

File: src/components/zen-modal/zen-modal.scss (hunk 2 above)
Comment (chavlji) [reply src/components/zen-modal/zen-modal.scss:57]: It's not a breaking change. It's just an improvement by designers.

Comment (chavlji) [reply src/components/zen-modal/zen-modal.scss:57]: Agree with that yeah. Noted!

**Pull request merged**